modules/lotameIdSystem.js

```diff
@@ -0,0 +1,178 @@
+    ).toUTCString();
+    storage.setCookie(PROFILE_ID_NAME, profileId, expirationDate, 'Lax', undefined, undefined);
+  }
+  if (storage.hasLocalStorage()) {
+    storage.setDataInLocalStorage(PROFILE_ID_NAME, profileId, undefined);
+  }
+}
+
+function getFirstPartyId() {
+  if (storage.cookiesAreEnabled()) {
+    return storage.getCookie(PROFILE_ID_NAME, undefined);
+  }
+  if (storage.hasLocalStorage()) {
+    return storage.getDataFromLocalStorage(PROFILE_ID_NAME, undefined);
+  }
+}
+
+function getFromStorage(key) {
+  return storage.getCookie(key) || storage.getDataFromLocalStorage(key);
+}
+
+function saveLotameCache(key, value) {
+  if (key && value) {
+    if (storage.cookiesAreEnabled()) {
+      let expirationDate = new Date(
+        utils.timestamp() + DAYS_TO_CACHE * DAYS_IN_MILLISECONDS
+      ).toUTCString();
+      storage.setCookie(key, value, expirationDate, 'Lax', undefined, undefined);
+    }
+    if (storage.hasLocalStorage()) {
+      storage.setDataInLocalStorage(key, value, undefined);
+    }
+  }
+}
+
+function getLotameLocalCache() {
+  let cache = {
+    lastUpdate: new Date(0),
+    data: getFromStorage(KEY_ID),
+    bundle: {
+      refreshSeconds: 3600,
+    },
+  };
+
+  try {
+    const rawBundle = getFromStorage(KEY_EXPIRY);
+    if (utils.isStr(rawBundle)) {
+      cache.bundle = JSON.parse(rawBundle);
+    }
+  } catch (error) {
+    utils.logError(error);
+  }
+  try {
+    const rawDate = getFromStorage(KEY_LAST_UPDATE);
+    if (utils.isStr(rawDate)) {
+      cache.lastUpdate = new Date(rawDate);
+    }
+  } catch (error) {
+    utils.logError(error);
+  }
+  return cache;
+}
+
+function clearLotameCache(key) {
+  if (key) {
+    if (storage.cookiesAreEnabled()) {
+      let expirationDate = new Date(0).toUTCString();
+      storage.setCookie(key, '', expirationDate, 'Lax', undefined, undefined);
+    }
+    if (storage.hasLocalStorage()) {
+      storage.removeDataFromLocalStorage(key, undefined);
+    }
+  }
+}
+
+/** @type {Submodule} */
+export const lotameIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+
+  /**
+   * Decode the stored id value for passing to bid requests
+   * @function decode
+   * @param {(Object|string)} value
+   * @returns {(Object|undefined)}
+   */
+  decode(value, configParams) {
+    return value;
+  },
+
+  /**
+   * Retrieve the Lotame id
+   * @function
+   * @param {SubmoduleParams} [configParams]
+   * @param {ConsentData} [consentData]
+   * @param {(Object|undefined)} cacheIdObj
+   * @returns {IdResponse|undefined}
+   */
+  getId(configParams, consentData, cacheIdObj) {
+    let localCache = getLotameLocalCache();
+
+    let refreshNeeded = false;
+    if (!utils.isStr(localCache.data)) {
+      const lastUpdate = localCache.lastUpdate;
+      refreshNeeded = (Date.now() - lastUpdate.getTime()) > localCache.bundle.refreshSeconds * 1000;
+    }
+
+    if (!refreshNeeded) {
+      return {};
```

Comment:
if a refresh isn't needed we return nothing instead of the id?

Reply:
Yes, it will test the user id module to check the stored value instead...correction: I was confusing it with the path I was going down yesterday...it need to return the id